internal/scheduler/legacyscheduler.go

```diff
@@ -0,0 +1,399 @@
+			return nil, err
+		}
+		jobSchedulingReport.PodSchedulingReports = append(jobSchedulingReport.PodSchedulingReports, report)
+
+		// Could not find a node for this pod.
+		if report.Node == nil {
+			jobSchedulingReport.UnschedulableReason = "failedToSchedulePod"
+			continue
+		}
+
+		// The job can be scheduled.
+		leasedJobsByQueue[queue] = append(leasedJobsByQueue[queue], candidateJob)
+
+		// Update the resource accounting for this queue.
+		totalResourcesByQueue[queue] = totalResourcesForQueue
+		roundResourcesByQueue[queue] = roundResourcesForQueue
+		roundResources = roundResourcesCopy
+
+		fmt.Println("queue resource usage: ", totalResourcesByQueue[queue])
+		consecutiveIterationsWithNoJobsLeased = 0
+		numJobsToLease += 1
+
+		// Exit if we've processed all jobs for some queue.
+		if len(jobCacheByQueue[queue]) == 0 && gotAllQueuedJobsByQueue[queue] {
+			break
+		}
+	}
+
+	jobs := make([]*api.Job, 0)
+	for queue, jobsToLease := range leasedJobsByQueue {
+
+		// TryLeaseJobs returns a list of jobs that were successfully leased.
+		// For example, jobs concurrently leased to another executor are skipped.
+		//
+		// TODO: Reports generated above will be incorrect if creating the lease fails.
+		successfullyLeasedJobs, err := c.JobQueue.TryLeaseJobs(c.ExecutorId, queue, jobsToLease)
+		if err != nil {
+			logging.WithStacktrace(log, err).Error("failed to lease jobs")
+		}
+		jobs = append(jobs, successfullyLeasedJobs...)
+	}
+
+	return jobs, nil
+}
+
+func WeightsFromAggregatedUsageByQueue(resourceScarcity map[string]float64, priorityFactorByQueue map[string]float64, aggregateResourceUsageByQueue map[string]schedulerobjects.ResourceList) map[string]float64 {
+	shares := make(map[string]float64)
+	for queue, rl := range aggregateResourceUsageByQueue {
+		priorityFactor, ok := priorityFactorByQueue[queue]
+		if !ok {
+			priorityFactor = 1
+		}
+		shares[queue] = priorityFactor / (ResourceListAsWeightedApproximateFloat64(resourceScarcity, rl) + 1)
+	}
+	return shares
+}
+
+func ResourceListAsWeightedApproximateFloat64(resourceScarcity map[string]float64, rl schedulerobjects.ResourceList) float64 {
+	usage := 0.0
+	for resourceName, quantity := range rl.Resources {
+		scarcity := resourceScarcity[resourceName]
+		// TODO: Why do we have our own Float64 conversion instead of quantity.AsApproximateFloat64?
+		usage += common.QuantityAsFloat64(quantity) * scarcity
+	}
+	return usage
+}
+
+// pickQueueRandomly returns a queue randomly selected from the provided map.
+// The probability of returning a particular queue AQueue is shares[AQueue] / sharesSum,
+// where sharesSum is the sum of all values in the provided map.
+func pickQueueRandomly(shares map[string]float64, seed int64) (string, float64) {
+	if len(shares) == 0 {
+		return "", 0
+	}
+
+	// Generate a random number between 0 and sum.
+	sum := 0.0
+	for _, share := range shares {
+		sum += share
+	}
+	pick := sum * rand.New(rand.NewSource(seed)).Float64()
```

Comment:
we should probably assert  sum isn't  zero here else we will panic later on.  I think the current caller of this function  won't ever pass a zero, but better to be safe than sorry!